types/react/test/tsx.tsx

```diff
@@ -122,3 +122,24 @@ export abstract class SetStateTestForAndedState<P, S> extends React.Component<P,
 		this.setState({ baseProp: 'foobar' });
 	}
 }
+
+interface NewProps { foo: string; }
+interface NewState { bar: string; }
+
+class ComponentWithNewLifecycles extends React.Component<NewProps, NewState> {
+    static getDerivedStateFromProps: React.GetDerivedStateFromProps<NewProps, NewState> = (nextProps) => {
```

Comment:
see my comment re: `Partial` vs `Pick`, if `NewState` had an additional member and this function only returns an object containing `bar` it will cause a compiler error

Reply:
Oops, that is correct, because the generic parameter is in the wrong part of the definition.

However, adding it to the correct side makes it not be inferred automatically and need to be specified 😱 

I guess I'll concede defeat and use `Partial` here. We can find out a better way to be stricter about this in a future PR.